framework/lib/components/tags-input/tags-input.tsx

```diff
@@ -0,0 +1,170 @@
+import React, {
+  FunctionComponent,
+  KeyboardEvent,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react'
+import {
+  ActionMeta,
+  CreatableSelect,
+  GroupBase,
+  InputActionMeta,
+  MultiValue,
+  OptionsOrGroups,
+} from 'chakra-react-select'
+import { equals, identity } from 'ramda'
+import { Box } from '../box'
+import { useToken } from '../../hooks'
+import { TagsInputProps } from './types'
+import { Option } from '../select'
+import { tagsInputStyles } from './tags-input-styles'
+
+/**
+ * Tags Input, based of react creatable select, is meant to select
+ * multiple tags. Follows styling of textarea and is resizable.
+ *
+ * @see {@link https://northlight.dev/reference/tags-input}
+ * @example (Example)
+ * ## Simple
+ * (?
+ * <TagsInput />
+ * ?)
+ *
+ * @example (Example)
+ * ## With suggestions
+ * (?
+ * <TagsInput options={[{ label: 'Option 1', value: '1' }, { label: 'Option 2', value: '2' }]}/>
+ * ?)
+ *
+ */
+export function TagsInput<T extends Option> ({
+  options,
+  onChange = identity,
+  isLoading,
+  loadingList = () => null,
+  'data-testid': testId,
+  value = [],
+  ...rest
+}: TagsInputProps<T>) {
+  const [ borderColor ] = useToken('colors.border.select', [ 'focus' ])
+  const [ menuIsOpen, setMenuIsOpen ] = useState(false)
+  const [ inputValue, setInputValue ] = useState('')
+  const [ selectedOptions, setSelectedOptions ] = useState<MultiValue<T>>(value)
+  const [ isFocused, setIsFocused ] = useState(false)
+  useEffect(() => {
+    setSelectedOptions(value)
+  }, [ value.length ])
+
+  const CustomSelect = CreatableSelect as FunctionComponent<TagsInputProps<T>>
+
+  const handleChange = (values: MultiValue<T>, actionMeta: ActionMeta<T>) => {
+    setSelectedOptions(values)
+    onChange(values, actionMeta)
+  }
+
+  const isValidNewOption = (input: string, availableOptions: MultiValue<T>) => {
+    const optionAlreadyExists = availableOptions.some(
+      (option) => option.value === input
+    )
+    return input !== '' && !optionAlreadyExists
+  }
+
+  const addNewOption = (newOption: T) => {
+    onChange(selectedOptions, { action: 'select-option', option: newOption })
+    setSelectedOptions((prev) => [ ...prev, newOption ])
+  }
+
+  const handleInputChange = (newInput: string, { action }: InputActionMeta) => {
+    setInputValue(newInput)
+    if (
+      !isValidNewOption(newInput, selectedOptions) ||
+      newInput === '' ||
+      newInput === ',' ||
+      !newInput.endsWith(',') ||
+      action === 'input-blur'
+    ) {
+      return
+    }
+    const newOption: Option = {
+      value: newInput,
+      label: newInput.slice(0, -1),
+    }
+    addNewOption(newOption as T)
+    setInputValue('')
+  }
+
+  const handleKeyDown = (event: KeyboardEvent<HTMLDivElement>) => {
+    if (
+      !isValidNewOption(inputValue, selectedOptions) &&
+      inputValue !== '' &&
+      event.key !== ' '
+    ) {
+      setInputValue('')
```

Comment:
i see this a lot, put it in a separate function: `clearInput` or something. That way, if we ever want to change the idea of `clear`, we'll have to modify only one single place.